redash/tasks/worker.py

```diff
@@ -50,8 +50,11 @@ def stop_executing_job(self, job):
         self.log.warning("Job %s has been cancelled.", job.id)
 
     def soft_limit_exceeded(self, job):
-        seconds_under_monitor = (utcnow() - self.monitor_started).seconds
-        return seconds_under_monitor > job.timeout + self.grace_period
+        if job.timeout < 0:
```

Comment:
Shouldn't it be `<=` ? Or zero timeout has some special meaning?

Reply:
Maybe we should use explicit -1 here? So it's clear what is it about? I would also consider defining a constant for it, so it's clear where this magic number coming from.

Reply:
@kravets-levko only -1 means "run infinitely". While I agree that a 0 doesn't make sense, someone might want to choose to set their Redash instance to time out immediately (🤷🏻‍♂️), so `soft_limit_exceeded` should include those. I agree with @arikfr's approach regarding explicitness.

Reply:
> I would also consider defining a constant for it, so it's clear where this magic number coming from.

Decided to avoid constant since it wouldn't be re-used in settings. Went for a [descriptive variable name](https://github.com/getredash/redash/pull/4653/commits/3e375bb528542e9c68cdce9036c328c71cac8dc0) instead.